experimental/defunct/test_device_seg_reduce.cu

```diff
@@ -1713,11 +1645,27 @@ struct DeviceSegReduceDispatch
         {
             // Get PTX version
             int ptx_version = 0;
-    #if (CUB_PTX_ARCH == 0)
-            if (CubDebug(error = PtxVersion(ptx_version))) break;
-    #else
-            ptx_version = CUB_PTX_ARCH;
-    #endif
+
+          // Define a temporary macro that expands to the current target ptx version
+          // in device code.
+          // <nv/target> may provide an abstraction for this eventually. For now,
+          // we have to keep this usage of __CUDA_ARCH__.
+#if defined(_NVHPC_CUDA)
+#define CUB_TEMP_GET_PTX __builtin_current_device_sm()
+#else
+#define CUB_TEMP_GET_PTX __CUDA_ARCH__
+#endif
+
+            NV_IF_TARGET(NV_IS_HOST, (
+                if (CubDebug(error = PtxVersion(ptx_version)))
```

Comment:
`PtxVersion` seems to encapsulate this logic. Can we remove `NV_IF_TARGET` here?

Reply:
You are correct :) Fixed.